content/en/docs/guides/deadlines.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Deadlines
+description: >-
+  Explains how deadlines can be used to effectively deal with unreliable
+  backends.
+---
+
+### Overview
+
+A deadline is used to specify a point in time past which a client is unwilling
+to wait for a response from a server. This simple idea is very important in
+building robust distributed systems. Clients that do not wait around
+unnecessarily and servers that know when to give up processing requests will
+improve the resource utilization and latency of your system.
+
+Note that while some language APIs have the concept of a __deadline__, others
+use the idea of a __timeout__. When an API asks for a deadline, you provide a
+point in time which the request should not go past. A timeout is the max
+duration of time that the request can take. For simplicity, we will only refer
+to deadline in this document.
+
+### Deadlines on the Client
+
+By default, gRPC does not set a deadline, which means it is possible for a
+client to end up waiting for a response effectively forever. To avoid this you
+should always explicitly set a realistic deadline in your clients. To determine
+the appropriate deadline you would ideally start with an educated guess based on
+what you know about your system (network latency, server processing time, etc.),
+validated by some load testing.
+
+If a server has gone past the deadline when processing a request, the client
+will give up and fail the RPC with the `DEADLINE_EXCEEDED` status. 
+
+### Deadlines on the Server
+
+A server might receive requests from a client with an unrealistically short
+deadline that would not give the server enough time to ever respond in time.
+This would result in the server just wasting valuable resources and in the worst
+case scenario, crash the server. A gRPC server deals with this situation by
+automatically cancelling a call (`CANCELLED` status) once a deadline set by the
+client has passed.
+
+Please note that the application is responsible for stopping any activity it has
```

Comment:
I'd specify "server application" instead of just "application" as even though it is clear in context some people my think of application as the client side.